### ISSUE TITLE
redundant btrfs swap subvol created

### ISSUE THREAD

**Issue by phillxnet** [OSInside/kiwi#1301]
## Problem description
Post #1297 and with the following config options, an unrequested / redundant swap subvol is created:
```
<type
    image="oem"
    initrd_system="dracut"
    filesystem="btrfs"
    installiso="true"
    bootpartition="false"
    devicepersistency="by-label"
    btrfs_root_is_snapshot="true"
...
    <systemdisk>
        <volume name="home"/>
        <volume name="root"/>
        <volume name="tmp"/>
        <volume name="opt"/>
        <volume name="srv"/>
        <volume name="boot/grub2/i386-pc"/>
        <volume name="boot/grub2/x86_64-efi"
                mountpoint="boot/grub2/x86_64-efi"/>
        <volume name="usr/local"/>
        <volume name="var" copy_on_write="false"/>
    </systemdisk>
...
<oemconfig>
    <oem-swap>true</oem-swap>
    <oem-swapsize>2048</oem-swapsize>

```
In the case of a btrfs swap subvol this may be miss-leading. I'm probably out of date but I seem to remember that swap on btrfs was only a fairly recently added capability. Thus it's creation could be confusing and it's utility/function would depend on the kernel version used. As a result I'm proposing this swap subvol should probably not be created unless specifically requested in the config.

## Expected behaviour
Only the requested or absolutely required for function btrfs subvols should be created. And given the potential question over btrfs's suitability / current compatibility re backing swap it might be better to not have this auto created.

## Steps to reproduce the behaviour
Build oem image type using the above configs and the below Kiwi version.
Resulting image does have dedicated working swap partition correctly configured (Thanks and much appreciated by the way).

Resulting btrfs subvol list:
```
btrfs subvolume list /
ID 256 gen 18 top level 5 path @
ID 258 gen 34 top level 256 path @/.snapshots
ID 259 gen 73 top level 258 path @/.snapshots/1/snapshot
ID 260 gen 47 top level 256 path @/home
ID 261 gen 74 top level 256 path @/opt
ID 262 gen 78 top level 256 path @/root
ID 263 gen 20 top level 256 path @/srv
ID 264 gen 13 top level 256 path @/swap
ID 265 gen 136 top level 256 path @/tmp
ID 266 gen 136 top level 256 path @/var
ID 267 gen 20 top level 256 path @/usr/local
ID 268 gen 24 top level 256 path @/boot/grub2/i386-pc
ID 269 gen 20 top level 256 path @/boot/grub2/x86_64-efi
ID 277 gen 31 top level 258 path @/.snapshots/2/snapshot

```
auto created partitions as expected (including the requested dedicated swap):
```
installer:~ # parted /dev/sda print
Model: ATA QEMU HARDDISK (scsi)
Disk /dev/sda: 25.8GB
Sector size (logical/physical): 512B/512B
Partition Table: gpt
Disk Flags: 

Number  Start   End     Size    File system     Name      Flags
 1      1049kB  3146kB  2097kB                  p.legacy  bios_grub
 2      3146kB  37.7MB  34.6MB  fat16           p.UEFI    boot, esp
 3      37.7MB  2185MB  2147MB  linux-swap(v1)  p.swap    swap
 4      2185MB  25.8GB  23.6GB  btrfs           lxroot
```
and successfully enabled:
```
installer:~ # cat /proc/swaps 
Filename                                Type            Size    Used    Priority
/dev/sda3                               partition       2097148 0       -1
```
via the newly fixed fstab entry of:
```
installer:~ # cat /etc/fstab 
LABEL=ROOT /.snapshots btrfs noatime,subvol=@/.snapshots 0 0
LABEL=ROOT /home btrfs noatime,subvol=@/home 0 0
LABEL=ROOT /opt btrfs noatime,subvol=@/opt 0 0
LABEL=ROOT /root btrfs noatime,subvol=@/root 0 0
LABEL=ROOT /srv btrfs noatime,subvol=@/srv 0 0
LABEL=ROOT /swap btrfs noatime,subvol=@/swap 0 0
LABEL=ROOT /tmp btrfs noatime,subvol=@/tmp 0 0
LABEL=ROOT /var btrfs noatime,subvol=@/var 0 0
LABEL=ROOT /usr/local btrfs noatime,subvol=@/usr/local 0 0
LABEL=ROOT /boot/grub2/i386-pc btrfs noatime,subvol=@/boot/grub2/i386-pc 0 0
LABEL=ROOT /boot/grub2/x86_64-efi btrfs noatime,subvol=@/boot/grub2/x86_64-efi 0 0
LABEL=ROOT / btrfs noatime 1 1
LABEL=SWAP swap swap defaults 0 0
LABEL=EFI /boot/efi vfat defaults 0 0
```

## OS and Software information
+ KIWI version: 9.19.7-lp151.1.1 Vendor: obs://build.opensuse.org/Virtualization:Appliances
+ Operating system: Leap 15.1 running Kiwi and trivially modified JeOS Leap 15.1 target image.
+ OBS version: Not used.

**Comment by phillxnet** [comment]
I do have some Python and shell capabilities and may have a little time to look to his issue, if it is accepted as such. But given my time constraints a pointer to where I should look in the code to 'fix' this issue, and potential pit-falls, would be much appreciated.

**Comment by davidcassany** [comment]
@phillxnet thanks much for you offer, but while investigating it I already implemented the solution. It was quite straight forward.

In any case i would be really grateful if you could give it check and validate this solves the reported issue.

**Comment by phillxnet** [comment]
@davidcassany Thanks for the reply and rapid fix, glad it was quite straight forward. Much appreciated.

I can confirm that with the exact same Kiwi config detailed as reproducer that this issue is now fixed.

Buildbot logs from un-patched / pre #1302:
KIWI version: 9.19.7-lp151.1.1 Vendor: obs://build.opensuse.org/Virtualization:Appliances
```
[ INFO    ]: 17:10:03 | --> creating EFI CSM(legacy bios) partition
[ INFO    ]: 17:10:05 | --> creating EFI partition
[ INFO    ]: 17:10:07 | --> creating SWAP partition
[ INFO    ]: 17:10:09 | --> creating root partition
[ INFO    ]: 17:10:11 | Creating EFI(fat16) filesystem on /dev/mapper/loop0p2
[ INFO    ]: 17:10:11 | Creating btrfs sub volumes
[ INFO    ]: 17:10:11 | --> sub volume home
[ INFO    ]: 17:10:11 | --> sub volume opt
[ INFO    ]: 17:10:11 | --> sub volume root
[ INFO    ]: 17:10:11 | --> sub volume srv
[ INFO    ]: 17:10:11 | --> sub volume swap
[ INFO    ]: 17:10:11 | --> sub volume tmp
[ INFO    ]: 17:10:11 | --> sub volume var
[ INFO    ]: 17:10:11 | --> setting no-copy-on-write for var
[ INFO    ]: 17:10:11 | --> sub volume usr/local
[ INFO    ]: 17:10:11 | --> sub volume boot/grub2/i386-pc
[ INFO    ]: 17:10:12 | --> sub volume boot/grub2/x86_64-efi
[ INFO    ]: 17:10:12 | Creating config.partids in boot system
```
And post patching my rpm installed instance of Kiwi by editing the following file as per #1302:
/usr/lib/python3.6/site-packages/kiwi/xml_state.py
I have the following:
```
[ INFO    ]: 12:27:55 | --> creating EFI CSM(legacy bios) partition
[ INFO    ]: 12:27:57 | --> creating EFI partition
[ INFO    ]: 12:27:59 | --> creating SWAP partition
[ INFO    ]: 12:28:01 | --> creating root partition
[ INFO    ]: 12:28:03 | Creating EFI(fat16) filesystem on /dev/mapper/loop0p2
[ INFO    ]: 12:28:03 | Creating btrfs sub volumes
[ INFO    ]: 12:28:03 | --> sub volume home
[ INFO    ]: 12:28:03 | --> sub volume opt
[ INFO    ]: 12:28:03 | --> sub volume root
[ INFO    ]: 12:28:04 | --> sub volume srv
[ INFO    ]: 12:28:04 | --> sub volume tmp
[ INFO    ]: 12:28:04 | --> sub volume var
[ INFO    ]: 12:28:04 | --> setting no-copy-on-write for var
[ INFO    ]: 12:28:04 | --> sub volume usr/local
[ INFO    ]: 12:28:04 | --> sub volume boot/grub2/i386-pc
[ INFO    ]: 12:28:04 | --> sub volume boot/grub2/x86_64-efi
[ INFO    ]: 12:28:04 | Creating config.partids in boot system
```
So post pr no 'swap' btrfs subvol creation logged.

Confirmed via the resulting installed image:
```
btrfs subvol list /
ID 256 gen 17 top level 5 path @
ID 258 gen 32 top level 256 path @/.snapshots
ID 259 gen 47 top level 258 path @/.snapshots/1/snapshot
ID 260 gen 47 top level 256 path @/home
ID 261 gen 46 top level 256 path @/opt
ID 262 gen 48 top level 256 path @/root
ID 263 gen 19 top level 256 path @/srv
ID 264 gen 51 top level 256 path @/tmp
ID 265 gen 51 top level 256 path @/var
ID 266 gen 19 top level 256 path @/usr/local
ID 267 gen 23 top level 256 path @/boot/grub2/i386-pc
ID 268 gen 19 top level 256 path @/boot/grub2/x86_64-efi
ID 276 gen 30 top level 258 path @/.snapshots/2/snapshot
```
The outputs of:
```
parted /dev/sda print
cat /proc/swaps 
```
are unaffected so we still have a configured and functioning swap partition.

Fstab now no longer has the prior 'swap' subvolume entry:
```
cat /etc/fstab 
LABEL=ROOT /.snapshots btrfs noatime,subvol=@/.snapshots 0 0
LABEL=ROOT /home btrfs noatime,subvol=@/home 0 0
LABEL=ROOT /opt btrfs noatime,subvol=@/opt 0 0
LABEL=ROOT /root btrfs noatime,subvol=@/root 0 0
LABEL=ROOT /srv btrfs noatime,subvol=@/srv 0 0
LABEL=ROOT /tmp btrfs noatime,subvol=@/tmp 0 0
LABEL=ROOT /var btrfs noatime,subvol=@/var 0 0
LABEL=ROOT /usr/local btrfs noatime,subvol=@/usr/local 0 0
LABEL=ROOT /boot/grub2/i386-pc btrfs noatime,subvol=@/boot/grub2/i386-pc 0 0
LABEL=ROOT /boot/grub2/x86_64-efi btrfs noatime,subvol=@/boot/grub2/x86_64-efi 0 0
LABEL=ROOT / btrfs noatime 1 1
LABEL=SWAP swap swap defaults 0 0
LABEL=EFI /boot/efi vfat defaults 0 0
```
A subsequent reboot of the installed system maintains the above configuration. Although we do still employ a one shot "dracut -f" systemd service.

Thanks again for all your time and effort on this project.
